ignition/src/GazeboWrapper.cpp

```diff
@@ -148,6 +171,12 @@ bool GazeboWrapper::findAndLoadSdf(const std::string& sdfFileName, sdf::Root& ro
         return false;
     }
 
+
+    // Configure SDF to fetch assets from ignition fuel.
+    sdf::setFindCallback(myFetchResource);
+    ignition::common::addFindFileURICallback(myFetchResourceUri);
+
+
```

Comment:
Now that I reached the end of the file I think I finally understood why you needed those methods and why you needed them as static. I suspect that you didn't think about allowing users from their C++ code to download resources from fuel bu calling `myGazeboWrapperObject.myFetchResource(nameOfFuelModel)`. You just needed some static function to be fed as callbacks. Did I get it right?

In this case, I think you can define the `myFetchResource*` methods as lambda functions directly here. This solution would avoid having them in the public APIs and it would fix the fetching problem. Can you please try using lambda functions instead?